product/views.py

```diff
@@ -0,0 +1,134 @@
+
+                    # new 상품일 경우 season 변수에 "NEW" 저장
+                    if prod['is_new']:
+                        season = "NEW"
+                    else:
+                        season = ""
+
+                    image_url_list = prod_image_url
+                    image_url_list = image_url_list.replace("[", "")
+                    image_url_list = image_url_list.replace("]", "")
+                    image_url_list = image_url_list.replace("'", "")
+                    image_url_list = image_url_list.split(",")
+
+                    prod = {
+                        "name"  : prod_item.name,
+                        "img"   : image_url_list,
+                        "season": season,
+                        "url"   : prod_url,
+                    }
+                    item[c_name].append(prod)
+        return JsonResponse(body, status = 200)
+
+class ProductCategoryListView(View):
+    def get(self, request, category_name):
+
+        # name_categorys = set()
+        # for product in products:
+        #     prod_name = product.name
+        #     prod_name = prod_name.name
+        #     split_names = prod_name.split()
+        #     prod_name = split_names[0] + " " + split_names[1]
+        #     name_categorys.add(prod_name)
+        data = ["CELINE", "CELINE WOMEN", "SHOES"]
+        products = Product.objects.all()
+        categorys = SubCategory.objects.all()
+
+        for cat in categorys:
+            if category_name == cat.name:
+                body = {
+                    "data"      : data,
+                    "item_list" : []
+                    }
+                item = {
+                    "category" : cat.name,
+                    "products" : []
+                    }
+                body["item_list"].append(item)
+                item2 = {
+                    "subcategory" : "",
+                    "list1"       : []
+                    }
+
+                item["products"].append(item2)
+
+                for prod in products:
+                    prod_name = prod.name
+                    prod_url = prod.product_url
+                    prod_image_url = prod.image_url
+                    image_url_list = prod_image_url.image_url
+                    image_url_list = image_url_list.replace("[", "")
+                    image_url_list = image_url_list.replace("]", "")
+                    image_url_list = image_url_list.replace("'", "")
+                    image_url_list = image_url_list.split(",")
+
+                    list = {
+                        "name"   : prod_name.name,
+                        "img"    : image_url_list,
+                        "is_new" : prod.is_new,
+                        "url"    : prod_url,
+                    }
+                    item2["list1"].append(list)
+        return JsonResponse(body, status = 200)
+
+class ProductDetailView(View):
+    def get(self, request, product_id):
+        if Product.objects.filter(id = product_id).exists():
```

Comment:
위와 같이 ORM을 사용하실 경우에,
`.exists()` 할 때, db hit 1번, `.get()` 할 때, db hit 1번 `총 2번 ` db에서 data를 조회하게 됩니다.

```suggestion
        try:
            product = Product.objects.get(id=product_id).select_related('image_url')
            result = {
                "id" : product.id,
                "name" : product.name,
                "description" : product.description,
                "price" : product.price,
                "color" : product.color,
                "image_urls" : product.image_url.replace("[", "").replace("]", "")
            }
            return JsonResponse(result, status=200) 
        except Product.DoesNotExist:
            return JsonResponse({"MESSAGE" : "IVALED_HOST"}, status=404) 
                
```